### PR TITLE
INTLY-1301 Add namespace params for backend and encryption secrets

### PR DESCRIPTION
**Pull request by aidenkeating**
* Use the namespace param for the component secret
* Introduce namespace params for backend and encryption secrets

### CODE REVIEW

**Comment by aidenkeating** [comment]
@pb82 @matskiv Mind taking a look? Verification is basically overriding the values and ensuring they're reflected when the Job tries to get the secret credentials etc. from a namespace

**Comment by matskiv** [comment]
👀

File: image/tools/entrypoint.sh [hunk 1]
```diff
@@ -12,9 +11,6 @@ while getopts "c:n:b:e:d:" opt; do
     c)
         component="$OPTARG"
         ;;
-    n)
-        component_secret_namespace="$OPTARG"
-        ;;
```
Comment (matskiv) [inline image/tools/entrypoint.sh:17]: `-n` CLI param needs to be removed from the templates as well

**Comment by matskiv** [comment]
I am still verifying this, having some troubles due to RBAC, and I noticed we should probably also a param for the namespace value of RoleBinding here:
https://github.com/integr8ly/backup-container-image/blob/master/templates/openshift/rbac/role-binding.yaml#L10

**Review by matskiv — APPROVED** [review]
Verified on my PDS cluster.
I'll approve, but it would be good to get RoleBinding parametrized as part of this PR as well.

**Comment by aidenkeating** [comment]
@matskiv Makes sense, let me update

**Comment by aidenkeating** [comment]
@matskiv Updated to template the RoleBinding to allow the namespace to be overridden. As the names of the Role and ServiceAccount are hardcoded in their files I've left them hardcoded in the RoleBinding.

**Pull request merged**